bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/handler/VerisureSmartLockThingHandler.java

```diff
@@ -0,0 +1,356 @@
+                            + "($giid: String!, $deviceLabel: String!, $input: LockDoorInput!) {\n  " + operation
+                            + "(giid: $giid, deviceLabel: $deviceLabel, input: $input)\n}\n";
+                    smartLockJSON.setQuery(query);
+                    list.add(smartLockJSON);
+
+                    String queryQLSmartLockSetState = gson.toJson(list);
+                    logger.debug("Trying to set SmartLock state to {} with URL {} and data {}", operation, url,
+                            queryQLSmartLockSetState);
+
+                    int httpResultCode = session.sendCommand(url, queryQLSmartLockSetState, installationId);
+                    if (httpResultCode == HttpStatus.OK_200) {
+                        logger.debug("SmartLock status successfully changed!");
+                    } else {
+                        logger.warn("Failed to send command, HTTP result code {}", httpResultCode);
+                    }
+                } else {
+                    logger.warn("PIN code is not configured! It is mandatory to control SmartLock!");
+                }
+            }
+        }
+    }
+
+    private void handeAutoRelockResult(String url, String data, BigDecimal installationId, Command command) {
+        logger.debug("Trying to set Auto Relock {} with URL {} and data {}", command.toString(), url, data);
+        if (session != null) {
+            int httpResultCode = session.sendCommand(url, data, installationId);
+            if (httpResultCode == HttpStatus.OK_200) {
+                logger.debug("AutoRelock sucessfully changed to {}", command.toString());
+            } else {
+                logger.warn("Failed to send command, HTTP result code {}", httpResultCode);
+            }
+        }
+    }
+
+    private void handleAutoRelock(Command command) {
+        String deviceId = config.getDeviceId();
+        if (session != null) {
+            VerisureSmartLocks smartLock = (VerisureSmartLocks) session.getVerisureThing(deviceId);
+            if (smartLock != null) {
+                BigDecimal installationId = smartLock.getSiteId();
+                String csrf = session.getCsrfToken(installationId);
+                StringBuilder sb = new StringBuilder(deviceId);
+                sb.insert(4, "+");
+                String data;
+                String url = SMARTLOCK_AUTORELOCK_COMMAND;
+                if (command == OnOffType.ON) {
+                    data = "enabledDoorLocks=" + sb.toString()
+                            + "&doorLockDevices%5B0%5D.autoRelockEnabled=true&_doorLockDevices%5B0%5D.autoRelockEnabled=on&_csrf="
+                            + csrf;
+                    handeAutoRelockResult(url, data, installationId, command);
+                } else if (command == OnOffType.OFF) {
+                    data = "enabledDoorLocks=&doorLockDevices%5B0%5D.autoRelockEnabled=true&_doorLockDevices%5B0%5D.autoRelockEnabled=on&_csrf="
```

Comment:
You don't need to specify the deviceId here?

Reply:
No, for some reason the API does not use device ID for OFF.